src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -328,7 +329,9 @@ private string GenerateId()
                 //sanitize external RequestId as it may not be hierarchical. 
                 //we cannot update ParentId, we must let it be logged exactly as it was passed.
                 string parentId = ParentId[0] == RootIdPrefix ? ParentId : RootIdPrefix + ParentId;
-                if (parentId[parentId.Length - 1] != '.')
+
+                char lastChar = parentId[parentId.Length - 1];
+                if (lastChar != '.' && lastChar != '_')
```

Comment:
right :)

Activity has an Id property that is hierarchical and looks like `|Server1-5d183ab6-a000b421.1.8e2d4c28_1.`

'_' and '.' are used as delimiters that separate nodes of the Id. They have a slightly different meaning: '_' means that that parent of the activity came from external process, '.' means that parent in internal. This improves readability and gives users an idea where the parent came from.

The code above tries to fix external parent Id if does not have correct format.
And there is a corner case when parent Id ends with '_'. This is a corner case since in general, every time activity leaves process, it should be assigned with the new Id that will end with  the '.'. However there are cases in practice when it does not happen.

Before this change, '.' was appended after it so the Id looks like `|123.abc_.def_`: with '_.'. 
This is a very minor fix that slightly removes readability and avoids having two 	
consecutive delimiters. 

This is a really corner case and I don't think there is any contract on how this case should exactly be processed, so I don't think it worth testing